tests/Unit/Evolution/Systems/ScalarWave/Test_Equations.cpp

```diff
@@ -249,5 +250,95 @@ SPECTRE_TEST_CASE("Unit.Evolution.Systems.ScalarWave.UpwindFlux",
+      make_with_random_values<tnsr::i<DataVector, Dim, Frame::Inertial>>(
+          make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_plus_int = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_minus_int = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+
+  const auto n_dot_flux_pi_ext = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto n_dot_flux_phi_ext =
+      make_with_random_values<tnsr::i<DataVector, Dim, Frame::Inertial>>(
+          make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_plus_ext = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_minus_ext = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+
+  // package data needed for flux
+  ScalarWave::PenaltyFlux<Dim> flux_computer{};
+  flux_computer.package_data(make_not_null(&packaged_data_int),
+                             n_dot_flux_pi_int, n_dot_flux_phi_int, v_plus_int,
+                             v_minus_int, unit_normal_int);
+  flux_computer.package_data(make_not_null(&packaged_data_ext),
+                             n_dot_flux_pi_ext, n_dot_flux_phi_ext, v_plus_ext,
+                             v_minus_ext, unit_normal_ext);
+
+  Scalar<DataVector> normal_dot_numerical_flux_pi(pow<Dim>(num_pts), 0.0);
+  Scalar<DataVector> normal_dot_numerical_flux_psi(pow<Dim>(num_pts), 0.0);
+  tnsr::i<DataVector, Dim, Frame::Inertial> normal_dot_numerical_flux_phi(
+      pow<Dim>(num_pts), 0.0);
+  apply_numerical_flux(flux_computer, packaged_data_int, packaged_data_ext,
+                       make_not_null(&normal_dot_numerical_flux_pi),
+                       make_not_null(&normal_dot_numerical_flux_phi),
+                       make_not_null(&normal_dot_numerical_flux_psi));
+
+  CHECK(normal_dot_numerical_flux_psi ==
+        Scalar<DataVector>(pow<Dim>(num_pts), 0.0));
+  CHECK(normal_dot_numerical_flux_pi ==
+        Scalar<DataVector>(get(n_dot_flux_pi_int) +
+                           0.5 * (get(v_minus_int) - get(v_plus_ext))));
+
+  tnsr::i<DataVector, Dim> normal_dot_numerical_flux_phi_expected(
+      pow<Dim>(num_pts), 0.0);
+  for (size_t d = 0; d < Dim; ++d) {
+    normal_dot_numerical_flux_phi_expected.get(d) =
+        n_dot_flux_phi_int.get(d) -
+        0.5 * (unit_normal_int.get(d) * get(v_minus_int) +
+               unit_normal_ext.get(d) * get(v_plus_ext));
+  }
+  CHECK(normal_dot_numerical_flux_phi ==
+        normal_dot_numerical_flux_phi_expected);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.ScalarWave.PenaltyFlux",
+                  "[Unit][Evolution]") {
+  constexpr size_t num_pts = 5;
```

Comment:
(rename this to pts_per_dim if you are following my above suggestion)

Reply:
renamed